url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```
@@ -0,0 +1,714 @@
+        }
+
+        // type="IngegerRange min=0 max=10"
+        private static StdArgList INTEGERRANGE_ARGLIST = new StdArgList(new FixedArg[]{}, new OptionalArg[]{}, new NamedArg[]{NAMED_MIN, NAMED_MAX});
+
+        @ScriptyCommand(name="IntegerRange")
+        public static TypeTuple integerRangeType(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                aArgs = INTEGERRANGE_ARGLIST.guard(aArgs, aCtx);
+                int lMin = (Integer) aArgs[1];
+                int lMax = (Integer) aArgs[2];
+                return new TypeTuple(new IntegerRangeType(lMin, lMax), null);
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        @ScriptyCommand(name="Long")
+        public static TypeTuple longType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return LONG_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        @ScriptyCommand(name="Short")
+        public static TypeTuple shortType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return SHORT_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        @ScriptyCommand(name="String")
+        public static TypeTuple stringType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return STRING_TUPLE;
+            }
+            catch (ArgSpecException e)
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)